Assets/Scripts/Player/PlayerStateController.cs

```diff
@@ -204,11 +199,7 @@ public void SetState(PlayerStates state)
                 if (heldInteractable)
                 {
                     //Refund turret
-                    inventoryManager.ResourceAmount += ui.GetSelectedSegment().cost;
-
-                    RemoveInteractable(heldInteractable);
-                    Destroy(heldInteractable.gameObject);
-                    heldInteractable = null;
```

Comment:
I don't see this line (`heldInteractable = null;`) in `SellTower()` below, is this on purpose..? 🤔

Reply:
Indeed it is. It turns out, `heldInteractable` is and has been broken for a long time. Nothing except loot currently uses it, and carrying loot has been prone to bugs in playtesting, most likely due to this. I have already created a Trello card in the backlog when I discovered this while making `SellTower().`

Speaking of `SellTower()`, it should function just fine without `heldInteractable = null;`, as it is completly independent of that variable as of this pull request. 😎